drivers/stepper_control/zephyr_stepper_motion_control/stepper_motion_control_timing_source.h

```diff
@@ -7,32 +7,55 @@
 #define ZEPHYR_DRIVER_STEPPER_STEP_DIR_STEPPER_TIMING_SOURCE_H_
 
 #include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/counter.h>
+
+struct timing_source_config {
+	const struct device *counter;
+};
+
+struct timing_source_data {
+	const struct device *motion_control_dev;
+	uint64_t microstep_interval_ns;
+	struct k_work_delayable stepper_dwork;
+
+#ifdef CONFIG_ZEPHYR_STEPPER_MOTION_CONTROL_COUNTER_TIMING
+	struct counter_top_cfg counter_top_cfg;
+	bool counter_running;
+#endif /* CONFIG_ZEPHYR_STEPPER_MOTION_CONTROL_COUNTER_TIMING */
+};
 
 /**
  * @brief Initialize the stepper timing source.
  *
- * @param dev Pointer to the device structure.
+ * @param config Pointer to the timing source configuration structure.
+ * @param data Pointer to the timing source data structure.
  * @return 0 on success, or a negative error code on failure.
  */
-typedef int (*stepper_timing_source_init)(const struct device *dev);
+typedef int (*stepper_timing_source_init)(const struct timing_source_config *config,
+					  struct timing_source_data *data);
 
 /**
  * @brief Update the stepper timing source.
  *
- * @param dev Pointer to the device structure.
+ * @param config Pointer to the timing source configuration structure.
+ * @param data Pointer to the timing source data structure.
  * @param microstep_interval_ns Step interval in nanoseconds.
  * @return 0 on success, or a negative error code on failure.
  */
-typedef int (*stepper_timing_source_update)(const struct device *dev,
+typedef int (*stepper_timing_source_update)(const struct timing_source_config *config,
```

Comment:
the `stepper_timing_source_update` function can be actually completely replaced by the `stepper_timing_source_start` function or do i miss something?